pkg/controller/atlasdeployment/atlasdeployment_controller.go

```diff
@@ -52,13 +52,23 @@ import (
 // AtlasDeploymentReconciler reconciles an AtlasDeployment object
 type AtlasDeploymentReconciler struct {
 	watch.ResourceWatcher
-	Client           client.Client
-	Log              *zap.SugaredLogger
-	Scheme           *runtime.Scheme
-	AtlasDomain      string
-	GlobalAPISecret  client.ObjectKey
-	GlobalPredicates []predicate.Predicate
-	EventRecorder    record.EventRecorder
+	Client                      client.Client
+	Log                         *zap.SugaredLogger
+	Scheme                      *runtime.Scheme
+	AtlasDomain                 string
+	GlobalAPISecret             client.ObjectKey
+	GlobalPredicates            []predicate.Predicate
+	EventRecorder               record.EventRecorder
+	ObjectDeletionProtection    bool
+	SubObjectDeletionProtection bool
+}
+
+type reconciliation struct {
```

Comment:
I found it hard to follow the reconciliation logic with this new struct. It doesn't make sense to add a new struct that will carry `AtlasDeploymentReconciler` and its dependencies inside of their own "methods".

Reply:
I can refactor and nor make it a method, but the issue is that some pieces of code require a lot of context, that would be way too many parameters to pass ungrouped.

Reply:
You just moved the standalone functions to be pointer receivers of your new struct, you could also make them receivers of the reconciler itself. I also think tons of parameters in functions are a bad thing but you could group the parameters in a struct per function instead of a centralized struct that carries the parent and its dependencies.

Reply:
Ok, I will make that struct a deletion context struct only. All methods will return to the reconciler struct. Then I will just pass the deletion context struct as a first parameter of the method where appropriate, probably only in the new method.

Reply:
Done